src/Microsoft.CodeQuality.Analyzers/CSharp/QualityGuidelines/CSharpAvoidDuplicateElementInitialization.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpAvoidDuplicateElementInitialization : AvoidDuplicateElementInitialization
+    {
+        private static readonly SyntaxGenerator s_generator = SyntaxGenerator.GetGenerator(new AdhocWorkspace(), LanguageNames.CSharp);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            analysisContext.RegisterOperationAction(AnalyzeOperation, OperationKind.ObjectCreation);
+        }
+
+        private static void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var objectInitializer = (IObjectCreationOperation)context.Operation;
+            var semanticModel = context.Compilation.GetSemanticModel(objectInitializer.Syntax.SyntaxTree);
```

Comment:
Its unfortunate that `OperationAnalysisContext` does not expose the semantic model. Can you please file an issue on https://github.com/dotnet/roslyn/issues?

Reply:
Done: https://github.com/dotnet/roslyn/issues/23645